### PR TITLE
inline fixtures

### LINKED ISSUE #9: Test fixtures
Could all the test fixtures please be moved out of `test/` and be put in `test/fixtures`?

### PR DESCRIPTION
**Pull request by calvinmetcalf**
fixes #9

### CODE REVIEW

**Comment by calvinmetcalf** [comment]
whops and noticed we never check the sign type, added those checks and also fixed the formatting, all as separate commits, so we can pull out the check sign type into a separate pull if we want.

**Comment by dcousens** [comment]
@calvinmetcalf I extracted all the tests to be easily testable based on their fixtures.
Please verify and merge if you ACK.

File: test/index.js [hunk 1]
```diff
@@ -1,472 +1,112 @@
...
+    var mySign = myCrypto.createSign(f.scheme)
 
     var mySig = mySign.update(message).sign(priv)
     var nodeSig = nodeSign.update(message).sign(priv)
     t.notEqual(mySig.toString('hex'), nodeSig.toString('hex'), 'not equal sigs')
-    var myVer = myCrypto.createVerify(scheme)
-    var nodeVer = nodeCrypto.createVerify(scheme)
+
+    // FIXME: deterministic signatures?
+//    t.equals(mySig.toString('hex'), f.signature)
```
Comment (dcousens) [inline test/index.js]: are the EC signatures meant to be deterministic?

File: test/index.js [hunk 2]
```diff
@@ -1,472 +1,112 @@
...
+    var mySign = myCrypto.createSign(f.scheme)
+    var nodeSign = nodeCrypto.createSign(f.scheme)
     var mySig = mySign.update(message).sign(priv)
     var nodeSig = nodeSign.update(message).sign(priv)
+
     t.equals(mySig.length, nodeSig.length, 'correct length')
     t.equals(mySig.toString('hex'), nodeSig.toString('hex'), 'equal sigs')
-    var myVer = myCrypto.createVerify(scheme)
-    var nodeVer = nodeCrypto.createVerify(scheme)
+    t.equals(mySig.toString('hex'), f.signature)
```
Comment (dcousens) [inline test/index.js]: I guess the RSA signatures are deterministic?

Comment (calvinmetcalf) [inline test/index.js]: RSA is always deterministic, the DSA ones were originally non deterministic
but are now.

File: test/index.js (hunk 1 above)
Comment (calvinmetcalf) [inline test/index.js]: Our ecdsa ones are deterministic but node's are not

Comment (dcousens) [inline test/index.js]: Then why does uncommenting that line make the tests fail?

Comment (calvinmetcalf) [inline test/index.js]: How did you calculate it?

Comment (calvinmetcalf) [inline test/index.js]: Wait ecdsa might not be deterministic its handled by elliptical

**Comment by dcousens** [comment]
@calvinmetcalf I re-computed the signatures as suggested, all good now, not sure how I got the wrong ones initially.

**Comment by calvinmetcalf** [comment]
ok switched it so it uses the presense of a passphrase to trigger skipping in node 0.10 and added the check in both dsa and rsa tests, plus added tests for node 0.12 and iojs

**Comment by calvinmetcalf** [comment]
ok and fixed the tests so that the full tests run for 0.11, 0.12, and iojs 1.5.whatever

**Comment by dcousens** [comment]
ACK, good to merge

**Comment by calvinmetcalf** [comment]
3.0.1

**Pull request merged**